src/extension.ts

```diff
@@ -66,8 +66,10 @@ async function registerProvider(
 ): Promise<void> {
   const disposable = provider.setKubernetesProviderConnectionFactory({
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    create: (params: { [key: string]: any }, logger?: Logger, token?: CancellationToken) =>
-      createCluster(params, logger, minikubeCli, telemetryLogger, token),
+    create: (params: { [key: string]: any }, logger?: Logger, token?: CancellationToken) => {
+      if (!minikubeCli) throw new Error('minikube executable cannot be found');
```

Comment:
issue: it's not related to typechecking --> separate PR/enhancement

Reply:
yes it is necessary because the following functions take a string as minikubeCli, and here it can be undefined

https://github.com/podman-desktop/extension-minikube/blob/364b0bb265eed5510930be656b4637e263f171b6/src/create-cluster.ts#L23-L27

So we need to do something to assert it is